src/Datadog.Trace/Tagging/HttpTags.cs

```diff
@@ -26,6 +36,36 @@ internal class HttpTags : InstrumentationTags, IHasStatusCode
 
         public string HttpStatusCode { get; set; }
 
-        protected override IProperty<string>[] GetAdditionalTags() => HttpTagsProperties;
+        protected override IProperty<string>[] GetAdditionalTags()
```

Comment:
I'm not sure if I should be providing optimization feedback at this time, but this `GetAdditionalTags()` will be called each time serialization is done, so we lose performance by dynamically building it up each time versus fetching a static readonly property. What if you created your own `OTelHttpTags` implementation that only implemented the OTel conventions so this class remains untouched?

Reply:
Hmm, actually this suggestion causes several cascading changes because we expect to have an `out HttpTags` in the actual instrumentation classes. Let me revisit that thought

Reply:
What about creating an issue for it? It would be easier to compare code and performance against something already working.

Reply:
Before deferring this to an issue, can you investigate a new approach where you create an `OTelHttpTags` class that subclasses the existing `HttpTags`? I'd like to see if it's possible to solve this in this PR because Datadog will be sensitive to taking code changes that reduce performance and this might solve it before it becomes a problem. If this approach doesn't work, then yes we can address this concern in a separate GitHub issue

Reply:
@zacharycmontoya will do tomorrow 😉

Reply:
I imagine you can override `GetAdditionalTags` so you have a pre-built list similar to `HttpTagsProperties` but with the customized mapping. This would be less flexible than what you have here, but I think it's worth the trade-off because we would only compute it once.

Reply:
Addressed: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/63/commits/5ac1fa56a482960563d290033e299f382b09c621